### DISCUSSION TITLE
Time series and COGs on the same map in Streamlit

### DISCUSSION THREAD

**Discussion by jakubstaszel** [opengeos/leafmap#452] [Q&A] [answered]
Hello,

I am on my way with my master thesis, here is the app: https://malopolska-drought.streamlit.app and here is the development version of the code for my app: https://github.com/jakubstaszel/malopolska-drought-app/tree/malopolska-webapp.

I have one issue that I don't know how to solve. I need to combine 2 features of leafmap in one Streamlit app:
1. Time series,
2. COGs.

When it comes to using Leafmap in Streamlit, the only supported way is to use folium, right? If yes, then my problem is that when using folium, time series is not supported: https://github.com/opengeos/leafmap/blob/02c9ff37918a9caf087a573db46fa5a9be2eb9ef/leafmap/foliumap.py#L2860

Do you have any ideas how to solve this issue? My idea was to use streamlit slider, but I am rerendering the whole map every time slider value is changed. It can work fine, but it will look much nicer if I change only the displayed layer. Can I do so? Or maybe I should use completely different approach to this problem?

All help is much appreciated, it is my first time working with Streamlit and leafmap.

**Comment by giswqs** [comment, marked as answer]
Try out Solara. It supports ipywidgets and ipyleaflet. See https://giswqs-solara-geospatial.hf.space/leafmap
